### PR TITLE
add an ability to check that a Path has an executable bit set

### LINKED ISSUE #1961: Allow for the ability to check whether a Path has an executable bit set
It would be nice to be able to use click in the scenario where the user might pass in a path to an executable. In the case that a binary is produced in a folder of the same name, this would catch the scenario where the user passes the directory containing the executable instead of the correct `/some/long/path/programname/programname`

This is very straightforward to add: https://github.com/sielicki/click/commit/9cd5150da149fa2bff21208811deac6938acfece

### PR DESCRIPTION
**Pull request by sielicki**
- fixes #1961 

Checklist:

- [ ] Add tests that demonstrate the correct behavior of the change. Tests should fail without the change.
- [x] Add an entry in `CHANGES.rst` summarizing the change and linking to the issue.
- [x] Add `.. versionchanged::` entries in any relevant code docs.
- [ ] Run `pre-commit` hooks and fix any issues.
- [ ] Run `pytest` and `tox`, no tests failed.

tox failed but as far as I can tell it's unrelated to my code. I'm also not sure where pre-commit hooks are documented, but as far as I can tell this is formatted fine.

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -9,8 +9,8 @@ Unreleased
 Version 8.0.2
 -------------
 
-Unreleased
-
+-   An an argument to ``Path`` which mirrors writable/readable, but for
```
Comment (davidism) [inline CHANGES.rst:12]: This needs to go in the 8.1 section.

File: src/click/types.py [hunk 2]
```diff
@@ -768,6 +769,9 @@ class Path(ParamType):
         ``None``, keep Python's default, which is ``str``. Useful to
         convert to :class:`pathlib.Path`.
 
+    .. versionchanged:: 8.0.2
+        Added the `executable` parameter.
```
Comment (davidism) [inline src/click/types.py:773]: Use double backticks to format code.

Comment (sielicki) [reply src/click/types.py]: Fixed, sorry for the lack of care with that.

File: CHANGES.rst [hunk 3]
```diff
@@ -3,15 +3,14 @@
 Version 8.1.0
 -------------
 
-Unreleased
-
+-   An an argument to ``Path`` which mirrors writable/readable, but for
+    executable bits. :issue:`1961`
```
Comment (davidism) [inline CHANGES.rst:7]: Keep two lines before subsequent headers, this format is used throughout the docs.

File: CHANGES.rst (hunk 1 above)
Comment (sielicki) [reply CHANGES.rst]: Thanks, wasn't sure whether this could be minor or not.

File: CHANGES.rst [hunk 4]
```diff
@@ -3,15 +3,14 @@
...
 
 Version 8.0.2
 -------------
 
 Unreleased
 
-
```
Comment (davidism) [inline CHANGES.rst:14]: Don't remove intentional formatting.

Comment (sielicki) [reply CHANGES.rst]: I'm sloppy. I *think* I got it right this time. ;)

**Comment by sielicki** [comment]
@davidism is there anything I can help with to make sure this lands in the next milestone? If this PR needs to be redone to point at a pallets:next branch, let me know.

**Pull request merged**